[Docs] Replace Icon Font with SVG Icons

mbrookes opened the pull request:
<!-- Thanks so much for your PR, your contribution is appreciated! -->

- [x] PR has tests / docs demo, and is linted.
- [x] Commit and PR titles begin with [ComponentName], and are in imperative form: "[Component] Fix leaky abstraction".
- [x] Description explains the issue / use-case resolved, and auto-closes the related issue(s) (http://tr.im/vFqem).

Update the `next` docs site to use SVG Icons so that it works offline (independently of the MD Icon Font).

oliviertassinari commented on docs/site/src/demos/bottom-navigation/LabelBottomNavigation.js:
```diff
@@ -32,19 +36,19 @@ export default class LabelBottomNavigation extends Component {
           <BottomNavigationButton
             label="Recents"
-            icon={<span className="material-icons">restore</span>}
+            icon={<RestoreIcon />}
```
No luck here 
![capture d ecran 2016-12-30 a 23 13 40](https://cloud.githubusercontent.com/assets/3165635/21573524/bd4177ce-cee5-11e6-91f9-2f8083bedb96.png)

oliviertassinari replied:
Oups, it's broken:
![capture d ecran 2017-01-01 a 13 17 14](https://cloud.githubusercontent.com/assets/3165635/21581157/b4c76334-d024-11e6-9100-df01cb13e14f.png)

mbrookes replied:
Wait, what? That was fixed by removing `display: block`. 

What browser is that? It's fine here in Chrome and Safari, but BrowserStack doesn't seem to want to work so I can't test FF and IE.

You sure you pulled yesterday's commits?

mbrookes replied:
Oh, never mind - that's BottomNav not Tabs. I guess it's the same issue.

mbrookes replied:
It doesn't look like removing `display: block` is going to work as a quick fix like it did for Tabs...

This could take me a while - I'm a bit rusty on my CSS!

oliviertassinari commented on docs/site/src/demos/tabs/IconLabelTabs.js:
```diff
@@ -24,15 +27,15 @@ export default class IconLabelTabs extends Component {
         >
           <Tab
-            icon={<span className="material-icons">phone</span>}
+            icon={<PhoneIcon />}
```
No luck here
![capture d ecran 2016-12-30 a 23 17 28](https://cloud.githubusercontent.com/assets/3165635/21573559/314d747e-cee6-11e6-9bd8-49812aff6a73.png)

oliviertassinari commented on docs/site/index.html:
```diff
@@ -8,7 +8,6 @@
       content="width=device-width, initial-scale=1, user-scalable=0, maximum-scale=1, minimum-scale=1"
     >
     <link href="https://fonts.googleapis.com/css?family=Roboto:400,400italic,500,700" rel="stylesheet" type="text/css">
-    <link href="https://fonts.googleapis.com/icon?family=Material+Icons" rel="stylesheet">
```
Don't we need to keep it to show of how font icon works?

mbrookes replied:
Yeah, I couldn't decide whether to leave it in anticipation of `FontIcon`, or be clean about it and remove it until it's needed.

mbrookes commented:
> Still some components to fix

Bah, thanks - thought I got them all. 

Not sure what's going on with Tabs - an Icon should be an Icon. I'll take a look.

mbrookes commented:
Okay, I'd updated the docs site components and examples, but missed the obvious fact that some components use icons internally, so when I removed the Icon Font, those broke.

Tabs is fixed, with a resulting small change to vertical spacing.

I've updated the regression test baseline image, but should probably add regression tests for both icon types.

@oliviertassinari Please could you take a look at the failing tests?

oliviertassinari commented:
> @oliviertassinari Please could you take a look at the failing tests?

Eslint is complaining. Do you have access to the circle logs?

mbrookes commented:
> Eslint is complaining. 

Bah, yes, I need to ad lint to git as a pre-push hook! 😄 

But I actually meant the unit tests that I mentioned in chat.

> Do you have access to the circle logs?

Yes.

oliviertassinari commented on src/svg-icons/phone.js:
```diff
@@ -0,0 +1,17 @@
+let Phone = (props) => (
+  <SvgIcon {...props}>
+    <path d="M6.62 10.79c1.44 2.83 3.76 5.14 6.59 6.59l2.2-2.2c.27-.27.67-.36 1.02-.24 1.12.37 2.33.57 3.57.57.55 0 1 .45 1 1V20c0 .55-.45 1-1 1-9.39 0-17-7.61-17-17 0-.55.45-1 1-1h3.5c.55 0 1 .45 1 1 0 1.25.2 2.45.57 3.57.11.35.03.74-.25 1.02l-2.2 2.2z"
+    />
```
Consistency?

mbrookes replied:
Mmm, yeah, I had originally manually fixed all the eslint errors as I copied the icons across, then decided it was easier to ignore them - must have manually reverted that one rather than recopying, and missed an undo.

Gotta hand it to you for a thorough review!!!

oliviertassinari commented on src/svg-icons/add.js:
```diff
@@ -0,0 +1,16 @@
+import pure from 'recompose/pure';
+import SvgIcon from '../SvgIcon';
+
+let Add = (props) => (
```
*Side comment as I don't think it makes sense here:*
We are not longer using arrow function for the components but named function.
```jsx
function Add(props)
```
@nathanmarks I'm curious to know the motivation. From my end,
1. It makes migration to class component easier
2. We are not messing around with the context.

mbrookes replied:
Shall I modify them all?

oliviertassinari replied:
I think it's fine line this 👍 .

oliviertassinari replied:
My only concern would be with the convention we should use for filenames.

oliviertassinari commented on docs/site/src/demos/lists/FolderList.js:
```diff
@@ -23,11 +24,11 @@ function FolderList(props, context) {
       <List>
         <ListItem button>
-          <Avatar icon={<span className="material-icons">folder</span>} />
+          <Avatar icon={<FolderIcon />} />
```
I have the following warning:
> warning.js:36Warning: Failed prop type: Invalid prop `icon` supplied to `Avatar`, expected a ReactNode.
    in Avatar (created by FolderList)
    in FolderList (created by Demo)
    in div (created by Demo)
    in div (created by Demo)
    in Demo (created by MarkdownDocs)
    in div (created by MarkdownDocs)
    in MarkdownDocs (created by RouterContext)
    in div (created by AppContent)
    in AppContent (created by RouterContext)
    in div (created by AppFrame)
    in AppFrame (created by Connect(AppFrame))
    in Connect(AppFrame) (created by withWidth(Connect(AppFrame)))
    in EventListener (created by withWidth(Connect(AppFrame)))
    in withWidth(Connect(AppFrame)) (created by RouterContext)
    in RouterContext (created by Router)
    in ScrollBehaviorContext (created by Router)
    in Router (created by AppRouter)
    in AppRouter (created by App)
    in MuiThemeProvider (created by App)
    in App (created by Connect(App))
    in Connect(App)
    in Provider
    in AppContainer

mbrookes replied:
> I have the following warning:

Good spot. 

Any thoughts on the unit test failures?

mbrookes commented on src/svg-icons/wifi.js:
```diff
@@ -0,0 +1,16 @@
+  </SvgIcon>
+);
+Wifi = pure(Wifi);
+Wifi.displayName = 'Wifi';
```
Okay, I'll remove it from the icons I copied over manually, and we can update the generator script separately.

mbrookes commented:
@oliviertassinari I had a crack at fixing the BottomNav layout, but let me know if there's a better way. Otherwise, just those pesky test failures when you get a chance to take a look.

oliviertassinari commented on src/BottomNavigation/BottomNavigationButton.js:
```diff
@@ -114,10 +114,11 @@ export default class BottomNavigationButton extends Component {
-    const iconClassName = classNames(classes.icon,
-      isValidElement(iconProp) ? iconProp.props.className : null);
+    const iconClassName = classNames(classes.icon, isValidElement(iconProp) ?
+      iconProp.props.className :
```
I'm not sure to understand. What the motivation for moving the icon `className` to the parent `div`?

mbrookes replied:
It fixes the layout issue - `display: block` doesn't work when applied directly to the SVG icon, but it does when applied to wrapper div. I don't know why. Like I said above: 

> I had a crack at fixing the BottomNav layout, but let me know if there's a better way.

mbrookes commented:
> My only concern would be with the convention we use for the svg-icons' filenames.

What's the concern?

Could you have a look at the failing unit tests please when you get a chance? (Sorry to keep asking).

oliviertassinari commented:
> What's the concern?

It regarding the naming convention of the file. For instance, why using `src/svg-icons/star.js` over `src/svgIcons/Star.js`?

> Could you have a look at the failing unit tests please when you get a chance? (Sorry to keep asking).

I'm not sure what's counter intuitive from the failing error. Basically, he can't find `wrapper.find('span.material-icons');`
The PR now needs to be rebased 😬 .

mbrookes commented:
> For instance, why using src/svg-icons/star.js over src/svgIcons/Star.js?

Hmmm right - I think when we switched naming conventions way back, that got left out as they're automatically generated, and the script (and everything that consumes the generated components) simply never got updated. Lets put it on the to-do list as a separate PR for `next` and / or separately packaged icons.

> I'm not sure what's counter intuitive from the failing error.

I was thrown by the slightly vague error message "Error: Method “text” is only meant to be run on a single node. 0 found instead.", and didn't look past it to the lines causing the error... thanks for the pointer.

oliviertassinari commented on src/internal/SwitchBase.spec.js:
```diff
@@ -115,6 +115,7 @@ describe('<SwitchBase />', () => {
     });
 
     it('should not not be checked', () => {
+      // console.log(wrapper.debug());
```
To remove

mbrookes replied:
😊

oliviertassinari commented on src/BottomNavigation/BottomNavigationButton.js:
```diff
@@ -131,7 +132,9 @@ export default class BottomNavigationButton extends Component {
         onClick={this.handleChange}
       >
-        {icon}
+        <div className={iconClassName}>
```
We can remove that extra DOM node and the associated logic change with that diff:
```diff
    icon: {
      display: 'block',
+     margin: 'auto',
    },
```

mbrookes replied:
> margin: 'auto',

Much better. I should have thought to try that. 😊 

I also took the liberty of adding some whitespace to the dense block of multi-line `const`s for readability. If that goes against code conventions I'll revert it, but I think it's for the better.
